scripts/mh_windowfill.pl

```diff
@@ -0,0 +1,244 @@
+##############################################################################
+#
+# mh_windowfill.pl v1.00 (20151116)
+#
+# Copyright (c) 2015  Michael Hansen
+#
+# Permission to use, copy, modify, and distribute this software
+# for any purpose with or without fee is hereby granted, provided
+# that the above copyright notice and this permission notice
+# appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+# WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL
+# THE AUTHOR BE LIABLE FOR  ANY SPECIAL, DIRECT, INDIRECT, OR
+# CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+# LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
+# NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+#
+##############################################################################
+#
+# fill windows so scrolling starts bottom-up instead of top-down
+#
+# known issues:
+# 	- /CLEAR will reset to top-down
+#	- it is possible to confuse the script into not filling with a combination
+#	  of script load/unloads and window resizes. but it requires effort
+#
+# history:
+#	v1.00 (20151116)
+#		initial release
+#
+
+use v5.14.2;
+
+use strict;
+
+##############################################################################
+#
+# irssi head
+#
+##############################################################################
+
+use Irssi 20100403;
+use Irssi::TextUI;
+
+our $VERSION = '1.00';
+our %IRSSI   =
+(
+	'name'        => 'mh_windowfill',
+	'description' => 'fill windows so scrolling starts bottom-up instead of top-down',
+	'license'     => 'BSD',
+	'authors'     => 'Michael Hansen',
+	'contact'     => 'mh on IRCnet #help',
+	'url'         => 'irc://open.ircnet.net',
+);
+
+##############################################################################
+#
+# global variables
+#
+##############################################################################
+
+our $windowfill_running = 0;
+
+##############################################################################
+#
+# common support functions
+#
+##############################################################################
+
+sub trim_spacelike($)
```

Comment:
I like to have them to catch stupid mistakes in my coding, if they cause trouble i will remove them, but i see no point in removing them unless they are proven to cause issues with the script